packages/context/src/binding-filter.ts

```diff
@@ -9,11 +9,36 @@ import {BindingAddress} from './binding-key';
 /**
  * A function that filters bindings. It returns `true` to select a given
  * binding.
+ *
+ * **NOTE**: We keep the generic type `T` for backward compatibility. It
+ * represents the value type for matched bindings.
```

Comment:
Would you mind copying my longer explanation from https://github.com/strongloop/loopback-next/pull/2704/files#diff-07c537de62e910016440ad9289c6f53cR12?

I would also like to add `TODO(semver-major)` comment here, to make it easy for us to find breaking changes to include when making a new semver-major release.